cmd/tendermint/commands/lite.go

```diff
@@ -99,34 +104,91 @@ func runProxy(cmd *cobra.Command, args []string) error {
 
 	witnessesAddrs := strings.Split(witnessAddrsJoined, ",")
 
+	// if no witnesses were provided make sure we pass an empty string
+	if witnessAddrsJoined == "" {
+		witnessesAddrs = []string{}
+	}
+
 	db, err := dbm.NewGoLevelDB("light-client-db", home)
 	if err != nil {
 		return fmt.Errorf("can't create a db: %w", err)
 	}
 
+	if primaryAddr == "" {
+		var err error
+		primaryAddr, witnessesAddrs, err = checkForExistingProviders(db)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve primary or witness from db. Error: %w", err)
+		}
+		if primaryAddr == "" {
+			return errors.New("no primary address was provided nor found. Please provide a primary (using -p)." +
+				" Run the command: tendermint light --help for more information")
+		}
+	} else {
+		err := saveProviders(db, primaryAddr, witnessAddrsJoined)
+		if err != nil {
+			logger.Error("Unable to save primary and or witness addresses", "err", err)
+		}
+	}
+
+	trustLevel, err := tmmath.NewFraction(trustLevelStr)
+	if err != nil {
+		return fmt.Errorf("can't parse trust level: %w", err)
+	}
+
 	var c *light.Client
 	if trustedHeight > 0 && len(trustedHash) > 0 { // fresh installation
-		c, err = light.NewHTTPClient(
-			chainID,
-			light.TrustOptions{
-				Period: trustingPeriod,
-				Height: trustedHeight,
-				Hash:   trustedHash,
-			},
-			primaryAddr,
-			witnessesAddrs,
-			dbs.New(db, chainID),
-			light.Logger(logger),
-		)
+		if sequential {
+			c, err = light.NewHTTPClient(
+				chainID,
+				light.TrustOptions{
+					Period: trustingPeriod,
+					Height: trustedHeight,
+					Hash:   trustedHash,
+				},
+				primaryAddr,
+				witnessesAddrs,
+				dbs.New(db, chainID),
+				light.Logger(logger),
+				light.SequentialVerification(),
```

Comment:
I feel like we can do

if sequential {
  options = append(options, light.SequentialVerification())
} else {
  options = append(options, light.SkippingVerification(trustLevel))
}

instead of two big ifs